test/integration/targets/s3_bucket/roles/s3_bucket/tasks/missing.yml

```
@@ -0,0 +1,26 @@
+---
+- name: 'Attempt to delete non-existent buckets'
+  block:
+    # ============================================================
+    #
+    # While in theory the 'simple' test case covers this there are
+    # ways in which eventual-consistency could catch us out.
+    #
+    - name: 'Delete non-existstent s3_bucket (never created)'
+      s3_bucket:
+        name: '{{ bucket_name }}'
+        state: absent
+      register: output
+
+    - assert:
+        that:
+          - output is success
```

that's interesting. if it's not `success` it will simple re-run in the `always` section. That should also fail than - only in case there is no aws delay ...

---

Failures on deletion are always a weird edge case.  What you'll see is that the 'always' clause ignores the error, but ultimately there's not much better we can do.